### PR TITLE
Fix Batch Normalization inference behavior when virtual_batch_size is set

### PR DESCRIPTION
**Pull request by myaaaaaaaaa**
Band-aid fix for #15790

### CODE REVIEW

File: keras/layers/normalization/batch_normalization.py [hunk 1]
```diff
@@ -829,9 +832,13 @@ def call(self, inputs, training=None):
             original_shape = tf.concat(
                 [tf.constant([-1]), original_shape[1:]], axis=0
             )
+
+            expanded_shape = (
+                [self.virtual_batch_size, -1] if training_value else [-1, 1]
+            )
```
Comment (fchollet) [inline keras/layers/normalization/batch_normalization.py:838]: Therefore you can just do:

```
if training is True:
    expanded_shape = [self.virtual_batch_size, -1]
else:
    expanded_shape = [-1, 1]
```

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/layers/normalization/batch_normalization.py [hunk 2]
```diff
@@ -892,9 +899,6 @@ def _compose_transforms(scale, offset, then_scale, then_offset):
                 offset += then_offset
             return (scale, offset)
 
-        # Determine a boolean value for `training`: could be True, False, or
-        # None.
-        training_value = control_flow_util.constant_value(training)
```
Comment (myaaaaaaaaa) [inline keras/layers/normalization/batch_normalization.py:897]: Thanks for the review! I did try using `training` by itself initially, but it failed due to being a `tf.Tensor`, so I moved `training_value` up from this line. Presumably this is a workaround for some behavior in `_get_training_value()`, but that should probably be fixed in another PR by someone more familiar with this codebase than me.

**Review by fchollet — APPROVED** [review]
LGTM, thanks

**Comment by fchollet** [comment]
I am seeing test failures -- can you please take a look?

```
_v1_session_test(f, self, config, *args, **kwargs)
  File "keras/testing_infra/test_combinations.py", [line 427](keras/testing_infra/test_combinations.py?l=427&ws=tap-prod-presubmit/221835056&snapshot=2), in _v1_session_test
    f(test_or_class, *args, **kwargs)
  File "keras/layers/normalization/batch_normalization_test.py", [line 423](keras/layers/normalization/batch_normalization_test.py?l=423&ws=tap-prod-presubmit/221835056&snapshot=2), in test_basic_batchnorm_v2_input_shape_and_virtual_batch_size
    _ = norm(np.ones((1, 28, 28)), training=True)
AssertionError: "requested shape requires" does not match "Dimension size must be evenly divisible by 6272 but is 784 for '{{node batch_normalization_2/Reshape}} = Reshape[T=DT_DOUBLE, Tshape=DT_INT32](Const_1, batch_normalization_2/concat_1)' with input shapes: [1,28,28], [4] and with input tensors computed as partial shapes: input[1] = [8,?,28,28]."
```

**Comment by fchollet** [comment]
To note, the test failure is in V1 session mode, hence why it did not show up on CI. Sorry about that. One thing we could do is exclude the behavior from executing in V1 session mode, so that legacy code is not affected by the change.

**Comment by myaaaaaaaaa** [comment]
Looks like Reshape throws a different exception message in V1 mode, this should fix it

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Comment by fchollet** [comment]
This is getting blocked on some internal failures. I will patch it and merge.

**Comment by myaaaaaaaaa** [comment]
Thanks!
